### ISSUE TITLE
Implement PKCS authentication 

### ISSUE THREAD

**Issue by JavaJoeS** [eclipse-platform/eclipse.platform#728]
### Discussed in https://github.com/eclipse-platform/eclipse.platform/discussions/680

<div type='discussions-op-text'>

<sup>Originally posted by **JavaJoeS** September 13, 2023</sup>
Add PKCS12 and PKCS11 implementation.
See; https://github.com/eclipse-m2e/m2e-core/discussions/1538</div>

**Comment by JavaJoeS** [comment]
Almost all internet sites are secured, most publicly, some privatized requiring user x509 Certificates.  Eclipse needs to have the ability to utilize keystores and truststores to provide SSLcontext.

**Comment by JavaJoeS** [comment]
Looking to add UI for a PKI selection in package;
package org.eclipse.ui.internal.net.auth
Looks to me like the place to put it?

**Comment by mickaelistria** [comment]
Yes, this seems like the right package where to add authentication features.

**Comment by JavaJoeS** [comment]
I pulled a fork and began adding in some code.   Its going to be a lot of work!

**Comment by akurtakov** [comment]
Closing this one as there is clear action pending on contributor and discussions would better be handled in less places. https://github.com/eclipse-platform/eclipse.platform/pull/1716#issuecomment-2615949879
